### ISSUE TITLE
Test failures

### ISSUE THREAD

**Issue by susilehtola** [openmopac/MOPAC#5]
I am getting a lot of test failures on my machine.
```
50% tests passed, 57 tests failed out of 114

Total Test time (real) = 119.87 sec

The following tests FAILED:
	  1 - port (Failed)
	  8 - ters_preserved (Failed)
	 52 - key-ampersand (Failed)
	 53 - key-plus (Failed)
	 54 - key-plusplus (Failed)
	 55 - key-0SCF (Failed)
	 56 - key-1ELECTRON (Failed)
	 57 - key-1SCF (Failed)
	 58 - key-A0 (Failed)
	 59 - key-AIGIN (Failed)
	 60 - key-AIGOUT (Failed)
	 61 - key-ALLBONDS (Failed)
	 62 - key-ALLVEC (Failed)
	 63 - key-AM1 (Failed)
	 64 - key-ANGSTROMS (Failed)
	 65 - key-AUTOSYM (Failed)
	 66 - key-BIRADICAL (Failed)
	 68 - key-CdotIdot (Failed)
	 70 - key-CHARGE (Failed)
	 71 - key-CIS (Failed)
	 72 - key-CISD (Failed)
	 73 - key-CISDT (Failed)
	 74 - key-DISP (Failed)
	 75 - key-DOUBLET (Failed)
	 76 - key-EXCITED (Failed)
	 77 - key-INDO (Failed)
	 79 - key-MECI (Failed)
	 80 - key-MNDO (Failed)
	 81 - key-MNDOD (Failed)
	 83 - key-MS (Failed)
	 84 - key-NOANCI (Failed)
	 85 - key-NONET (Failed)
	 86 - key-NOSYM (Failed)
	 87 - key-OCTET (Failed)
	 88 - key-OPEN (Failed)
	 89 - key-PECI (Failed)
	 90 - key-PM3 (Failed)
	 91 - key-PM6 (Failed)
	 92 - key-PM6-D3 (Failed)
	 93 - key-PM6-D3H4 (Failed)
	 94 - key-PM6-D3H4X (Failed)
	 95 - key-PM6-DHplus (Failed)
	 96 - key-PM6-DH2 (Failed)
	 97 - key-PM6-DH2X (Failed)
	 98 - key-PM7 (Failed)
	100 - key-QUARTET (Failed)
	101 - key-QUINTET (Failed)
	102 - key-RHF (Failed)
	103 - key-RM1 (Failed)
	104 - key-ROOT (Failed)
	105 - key-SEPTET (Failed)
	106 - key-SEXTET (Failed)
	107 - key-SINGLET (Failed)
	108 - key-SNAP (Failed)
	109 - key-SYMMETRY (Failed)
	111 - key-TRIPLET (Failed)
	112 - key-UHF (Failed)
Errors while running CTest
```
It looks like a lot of these are caused by the removal of the TOTAL ENERGY entries from the output file, e.g.
```

          MOPAC Job: "ampersand.mop" ended normally on Dec  6, 2021, at 17:47.

Traceback (most recent call last):
  File "/home/susi/rpmbuild/BUILD/MOPAC-master/tests/keywords/../output_compare.py", line 238, in <module>
    assert ref == out, f'ERROR: string mismatch between {ref} and {out} on output line {line}'
AssertionError: ERROR: string mismatch between TOTAL and COSMO on output line 73
```

**Comment by godotalgorithm** [comment]
I had recently forgotten to commit my updates to the keyword tests, which caused most of these errors. I've corrected this problem.

However, this does not explain the failure of the "port" and "ters_preserved" tests. Can you post the output of those tests so that I can further diagnose the problem?

**Comment by susilehtola** [comment]
The error of `port` is
```
Test project /home/work/MOPAC/objdir
        Start   1: port
  1/114 Test   #1: port .............................***Failed   67.64 sec
          Thu Dec  9 09:48:14 2021  Job: 'port' started successfully


          MOPAC Job: "port.mop" ended normally on Dec  9, 2021, at 09:49.

WARNING: numerical mismatch between 378.82 and 378.83 on output line 72721
WARNING: numerical mismatch between 0.02506 and 0.02467 on output line 74051
WARNING: numerical mismatch between 0.01253 and 0.01233 on output line 74051
Traceback (most recent call last):
  File "/home/work/MOPAC/tests/output_compare.py", line 238, in <module>
    assert ref == out, f'ERROR: string mismatch between {ref} and {out} on output line {line}'
AssertionError: ERROR: string mismatch between LARGE and 1ELECTRON on output line 77509

```
`ters_preserved` fails with
```
  8/114 Test   #8: ters_preserved ...................***Failed    0.21 sec


          MOPAC Job: "Check_that_TERs_are_preserved.mop" ended normally on Dec  9, 2021, at 09:51.

Traceback (most recent call last):
  File "/home/work/MOPAC/tests/output_compare.py", line 238, in <module>
    assert ref == out, f'ERROR: string mismatch between {ref} and {out} on output line {line}'
AssertionError: ERROR: string mismatch between ** and GEO_DAT on output line 2

```
and finally, `key-ROOT` fails with
```
104/114 Test #104: key-ROOT .........................***Failed   15.37 sec


          MOPAC Job: "ROOT.mop" ended normally on Dec  9, 2021, at 09:53.

Traceback (most recent call last):
  File "/home/work/MOPAC/tests/keywords/../output_compare.py", line 238, in <module>
    assert ref == out, f'ERROR: string mismatch between {ref} and {out} on output line {line}'
AssertionError: ERROR: string mismatch between A1g and ???? on output line 106

```
The other tests are now clean.

**Comment by godotalgorithm** [comment]
I fixed the portability issues in the port and ters_preserved tests, and I'm not seeing any problems when I run the key-root test on Linux (on the GHA runner). Does the key-root test still fail for you? If so, I might need your assistance in debugging it. It is a very straightforward problem at least - the symmetry of the reference excited state is correctly stated in the list of excitations, but it isn't being copied correctly to the simulation summary (the ionization potential line).

**Comment by susilehtola** [comment]
Yes, key-root is still failing
```
Test project /home/work/MOPAC/objdir
    Start 104: key-ROOT
1/1 Test #104: key-ROOT .........................***Failed    9.18 sec


          MOPAC Job: "ROOT.mop" ended normally on Jan 13, 2022, at 20:18.

Traceback (most recent call last):
  File "/home/work/MOPAC/tests/keywords/../output_compare.py", line 238, in <module>
    assert ref == out, f'ERROR: string mismatch between {ref} and {out} on output line {line}'
AssertionError: ERROR: string mismatch between A1g and ???? on output line 106


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   9.18 sec

The following tests FAILED:
	104 - key-ROOT (Failed)
Errors while running CTest
```

**Comment by godotalgorithm** [comment]
I was able to reproduce this bug (using gcc on Linux) and subsequently find and fix it: eabcc5998686b2e6b5dc36be03f0c5469cceeacc
